services/core/tests/unit/signals.test.js

```diff
@@ -0,0 +1,91 @@
+/*
+ * LiskHQ/lisk-service
+ * Copyright © 2021 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ *
+ */
+const signals = require('../../shared/signals');
+
+describe('Signals tests', () => {
+    describe('signal.dispatch()', () => {
+        const signal = signals.get('dispatchEvent');
+        it('Dispatch and add signal data', async () => {
+            signal.dispatch('Event is dispatched');
+            signal.add((data) => {
+                expect(data.toEqual('Event is dispatched'));
+            });
+        });
+
+        it.todo('Failing test case for dispatched event');
+    });
+
+    describe('signal.add()', () => {
+        const signal = signals.get('addEvent');
+        const testFunc = () => { };
+
+        it('Add valid listener to signal.add()', async () => {
+            // initially 0 listeners
+            expect(signal.getNumListeners()).toBe(0);
+
+            // Add 1 listener to signals
+            signal.add(testFunc);
+            expect(signal.getNumListeners()).toBe(1);
+        });
+
+        it('Add invalid listener to signal.add() throws an error', async () => {
+            expect(() => { signal.add(); }).toThrowError('listener is a required param of add() and should be a Function.');
```

Comment:
Please consider keeping it generic, so that the test do not require changes if the error messages get updated in the future version of `signal`.
```suggestion
            expect(() => signal.add()).toThrow();
```
or 
```suggestion
            expect(() => signal.add()).toThrowError(/add\(\)/);
```